pymc_experimental/distributions/timeseries.py

```diff
@@ -0,0 +1,198 @@
+            if not at.all(at.lt(x0, n_states - 1)).eval():
+                raise ValueError('At least one initial state is larger than the number of states in the Markov Chain')
+
+        elif not isinstance(x0.owner.op, Discrete):
+            raise ValueError('x0 must be a discrete distribution')
+
+        else:
+            x0_probs = x0.owner.inputs[-1].eval()
+            n_cats = 1 if x0_probs.ndim == 0 else len(x0_probs)
+
+            if not n_cats <= n_states:
+                raise ValueError(
+                    'x0 has support over a range of values larger than the number of states in the Markov Chain')
+
+        return super().dist([P, logit_P, steps, x0], **kwargs)
+
+    @classmethod
+    def rv_op(cls, P, logit_P, steps, x0, size=None):
+        if size is not None:
+            batch_size = size
+        else:
+            batch_size = at.broadcast_shape(x0)
+
+        x0_ = x0.type()
+        P_ = P.type()
+        steps_ = steps.type()
+
+        state_rng = pytensor.shared(np.random.default_rng())
+
+        def transition(previous_state, transition_probs, old_rng):
+            p = transition_probs[previous_state]
+            next_rng, next_state = pm.Categorical.dist(p=p, rng=old_rng).owner.outputs
+            return next_state, {old_rng: next_rng}
+
+        markov_chain, state_updates = pytensor.scan(transition,
+                                                  non_sequences=[P_, state_rng],
+                                                  outputs_info=[x0_],
+                                                  n_steps=steps_,
+                                                  strict=True)
+
+        (state_next_rng,) = tuple(state_updates.values())
+
+        discrete_mc_ = at.concatenate([x0_[None, ...], markov_chain], axis=0).dimshuffle(
+            tuple(range(1, markov_chain.ndim)) + (0,)
+        ).squeeze()
+
+        discrete_mc_op = DiscreteMarkovChainRV(
+            inputs=[P_, x0_, steps_],
+            outputs=[state_next_rng, discrete_mc_],
+            ndim_supp=1,
+        )
+
+        discrete_mc = discrete_mc_op(P, x0, steps)
+        return discrete_mc
+
+
+@_change_dist_size.register(DiscreteMarkovChainRV)
+def change_mc_size(op, dist, new_size, expand=False):
+    if expand:
+        old_size = dist.shape[:-1]
+        new_size = tuple(new_size) + tuple(old_size)
+
+    return DiscreteMarkovChainRV.rv_op(
+        *dist.owner.inputs[:-1],
+        size=new_size,
+    )
+
+
+@_logprob.register(DiscreteMarkovChainRV)
+def discrete_mc_logp(
+        op, values, P, x0, steps, state_rng, **kwargs
+):
+
+    (value,) = values
+    # GARCH11 swaps the time axis to the front, not sure why this is necessary
+    mc_logprob = at.log(P[value[..., :-1], value[..., 1:]]).sum(axis=-1)
+
+    return mc_logprob
+
+@_moment.register(DiscreteMarkovChainRV)
+def discrete_markov_chain_moment(
+        op, rv, P, x0, steps, state_rng
+):
+    # TODO: What is the mean of the chain?
+    return at.zeros_like(rv)
```

Comment:
Is that the only place the moment is used?

The only moment-like summary statistic I know for these markov chains is the stationary distribution. Would computing that be useful anywhere? I don't think so but I also have no idea.

Reply:
Might be useful for users but wouldn't be used internally anywhere no. It doesn't even matter for observed variables.

So definitely not a blocker

Reply:
Is it worth returning zeros just to silence the warning? All zeros is a valid markov chain under any parameterization.

Reply:
What about (Pt+1=0|Pt=0)=0?

Maybe you can just return rv? Maybe just set default `initval` in `__new__` to be `prior`? That should do the same without the warning

Reply:
>What about (Pt+1=0|Pt=0)=0?

In principal we should probably dis-allow absorbing states. I have no proof, but I suspect they would cause numerical headaches.

Reply:
This is not an absorbing state, it's a forced switch state. Eg `P=[[0, 0.5, 0.5], [1/3, 1/3, 1/3], [1/3, 1/3, 1/3]]`

Reply:
Yes you are absolutely correct! My mistake.

Reply:
Removed the moment method, added `initval='prior'` as a default argument to `__new__`.